libcontainer/user/user.go

```diff
@@ -179,15 +180,53 @@ func ParseGroupFilter(r io.Reader, filter func(Group) bool) ([]Group, error) {
 	if r == nil {
 		return nil, errors.New("nil source for group-formatted data")
 	}
+	rd := bufio.NewReader(r)
+	out := []Group{}
 
-	var (
-		s   = bufio.NewScanner(r)
-		out = []Group{}
-	)
+	// Read the file line-by-line.
+	for {
+		var (
+			isPrefix  bool
+			wholeLine []byte
+			err       error
+		)
 
-	for s.Scan() {
-		text := s.Text()
-		if text == "" {
+		// Read the next line. We do so in chunks (as much as reader's
+		// buffer is able to keep), check if we read enough columns
+		// already on each step and store final result in wholeLine.
+		for {
+			var line []byte
+			line, isPrefix, err = rd.ReadLine()
+
+			if err != nil {
+				// We should return no error if EOF is reached
+				// without a match.
+				if err == io.EOF { //nolint:errorlint // comparison with io.EOF is legit, https://github.com/polyfloyd/go-errorlint/pull/12
```

Comment:
Why not use errors.Is ?

Reply:
Many functions from https://golang.org/pkg/bufio/ (including `ReadLine` method) legitimately return `io.EOF` error as a way to notify about EOF. Surely it is possible to use `errors.Is` here as well but for cases like this (when we do know we have a moderate to high chances to get this particular error, and it will never be wrapped) this is redundant and a direct comparison is fine.

errorlint has a set of exclusions configured for such direct io.EOF comparisons, and since https://github.com/polyfloyd/go-errorlint/pull/12 this is not even reported by the linter.

golang stdlib itself uses direct comparison with io.EOF quite often, and never uses `errors.Is` with `io.EOF`.

```
[kir@kir-rhat src]$ grep -r 'err == io.EOF' | grep -v ^vendor/ | wc -l
186
[kir@kir-rhat src]$ grep -r 'errors.Is' | grep io.EOF | wc -l
0
```

(this is in Go 1.17beta1 sources)